src/app/js/characteristic/index.js

```diff
@@ -58,19 +55,56 @@ export default handleActions({
     }),
     UPDATE_CHARACTERISTICS_SUCCESS: (state, { payload: characteristics }) => ({
         ...state,
-        characteristics,
+        characteristics: [
+            characteristics,
+            ...state.characteristics,
+        ],
         newCharacteristics: characteristics,
         editing: false,
         error: null,
         updating: false,
     }),
 }, defaultState);
 
-export const getDatasetTitle = ({ characteristic: { characteristics } }) => {
-    const titleCharacteristic = characteristics.find(({ scheme }) => scheme === TITLE_SCHEME);
-    return titleCharacteristic ? titleCharacteristic.value : null;
+export const getDatasetTitle = ({ publication: { fields }, characteristic: { characteristics } }) => {
+    const titleCharacteristic = fields.find(({ cover, scheme }) =>
+        cover === COVER_DATASET && scheme === TITLE_SCHEME);
+
+    return titleCharacteristic && characteristics[titleCharacteristic.name]
+        ? characteristics[titleCharacteristic.name]
+        : null;
 };
 
+const getCharacteristics = state => state.characteristic.characteristics[0] || {};
+
+export const getCharacteristicsLastVersion = createSelector(
+    getCharacteristics,
+    getFields,
+    (characteristic, fields) => Object
```

Comment:
Map on fields instead og characteristics keys, this way it is the fields that determines the characteristics order